src/core/qgsrulebasedlabeling.cpp

```diff
@@ -451,3 +451,46 @@ bool QgsRuleBasedLabeling::requiresAdvancedEffects() const
 {
   return mRootRule->requiresAdvancedEffects();
 }
+
+void QgsRuleBasedLabeling::toSld( QDomNode &parent, const QgsStringMap &props ) const
+{
+  if ( mRootRule == nullptr )
+  {
+    return;
+  }
+
+  QgsRuleBasedLabeling::RuleList rules = mRootRule->children();
+  for ( RuleList::const_iterator it = rules.constBegin(); it != rules.constEnd(); ++it )
+  {
+    Rule *rule = *it;
```

Comment:
General input, this can be simplified to:

```cpp
const QgsRuleBasedLabeling::RuleList rules = mRootRule->children();
 for ( Rule *rule : rules )
 {
```

Reply:
Thanks, fixing it